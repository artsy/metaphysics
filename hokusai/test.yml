version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
