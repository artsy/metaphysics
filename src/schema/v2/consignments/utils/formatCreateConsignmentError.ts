import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
