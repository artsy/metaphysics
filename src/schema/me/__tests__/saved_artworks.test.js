import gql from "test/gql"
import { resolve } from "path"
import { readFileSync } from "fs"
import { runAuthenticatedQuery } from "test/utils"

describe("me { saved_artwork", () => {
  describe("Handles getting collection metadata", () => {
    it("returns artworks for a collection", async () => {
      const artworksPath = resolve("src", "test", "fixtures", "gravity", "artworks_array.json")
      const artworks = JSON.parse(readFileSync(artworksPath, "utf8"))

      const query = gql`
        {
          me {
            saved_artworks {
              description
              artworks_connection(first: 10) {
                edges {
                  node {
                    id
                    title
                  }
                }
              }
            }
          }
        }
      `
      const rootValue = {
        collectionLoader: id => id === "saved-artwork" && Promise.resolve({ description: "My beautiful collection" }),
        collectionArtworksLoader: params => {
          if (params === { size: 10, offset: 0, total_count: true }) {
            return Promise.resolve({ body: artworks, headers: { "x-total-count": 10 } })
          }
        },
      }
      const data = await runAuthenticatedQuery(query, rootValue)
      expect(data).toMatchSnapshot()
    })
  })
})
