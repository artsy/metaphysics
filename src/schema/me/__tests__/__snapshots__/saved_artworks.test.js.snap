// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`me { saved_artwork Handles getting collection metadata returns artworks for a collection 1`] = `
Object {
  "me": null,
}
`;
